SpotifySearch/SpotifySearch/AlbumFactory.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  AlbumFactory.swift
+//  SpotifySearch
+//
+//  Created by Edward Anchundia on 10/28/16.
+//  Copyright © 2016 Edward Anchundia. All rights reserved.
+//
+
+import Foundation
+
+class AlbumFactory {
+    
+    static let manager: AlbumFactory = AlbumFactory()
+    private init() {}
+    
+    internal func getAlbums(from jsonData: Data) -> [Album]? {
+        
+        do {
+            let albumJSONData: Any = try JSONSerialization.jsonObject(with: jsonData, options: [])
+            
+            guard let albumJSONCasted = albumJSONData as? [String : AnyObject],
+                let albumArray = albumJSONCasted["album"] as? [String: AnyObject],
+                let albumItems = albumArray["item"] as? [[String: Any]] else {
```

Comment:
You have to read the JSON data carefully. One mistake and your complete guard statement will fail which is what happened here. The correct keys are `albums`, `items`, `url`. If you fix these you'll see you'll be able to pull data down. 

`
guard let albumJSONCasted = albumJSONData as? [String : AnyObject],
                let albumArray = albumJSONCasted["albums"] as? [String: AnyObject],
                let albumItems = albumArray["items"] as? [[String: Any]] else {
                    return nil
            }
            
            var albums: [Album] = []
            
            albumItems.forEach { albumObject in
                
                if let name = albumObject["name"] { ... }
`